core/test/reorder/rcm.cpp

```diff
@@ -33,6 +33,17 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/reorder/rcm.hpp>
 
 
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
The documentation on these pragmas is not very clear. I'm unsure how leaving out the corresponding `pop` effects warnings for other files. But, I would guess it doesn't affect those.